sprocs/question_order.sql

```diff
@@ -1,8 +1,33 @@
+DROP FUNCTION IF EXISTS question_order(BIGINT);
 CREATE OR REPLACE FUNCTION
     question_order (
         assessment_instance_id BIGINT
-    ) RETURNS TABLE (instance_question_id BIGINT, row_order INTEGER, question_number TEXT)
+    ) RETURNS TABLE (
+        instance_question_id BIGINT,
+        row_order INTEGER,
+        question_number TEXT,
+        sequence_locked BOOLEAN
+    )
 AS $$
+-- Used to determine if an instance question should block 
+-- access to further questions when minAdvancePerc is set.
+WITH locks_next AS (
+    SELECT 
+        iq.id AS instance_question_id,
+        NOT ( --- Do not lock if:
+            (iq.open = false) -- Run out of attempts on previous question
+            OR (aq.effective_min_advance_perc = 0) -- Zero percent required to advance
+            -- Score on previous question >= its unlock score
```

Comment:
When we are computing `locks_next`, we aren't concerned with the previous question yet.

```suggestion
            -- Score >= unlock score
```

Reply:
True though!